File: finance-backend/src/main/java/rooty/finance/financebackend/api/AnalyticsController.java
package rooty.finance.financebackend.api;

import java.time.LocalDate;
import java.time.YearMonth;
import java.util.List;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;
import rooty.finance.financebackend.api.dto.CategoryAmountDto;
import rooty.finance.financebackend.api.dto.MonthSummaryDto;
import rooty.finance.financebackend.api.dto.NetWorthPointDto;
import rooty.finance.financebackend.service.AnalyticsService;

@RestController
@RequestMapping("/api/analytics")
public class AnalyticsController {

    private final AnalyticsService analyticsService;

    public AnalyticsController(AnalyticsService analyticsService) {
        this.analyticsService = analyticsService;
    }

    @GetMapping("/month-summary")
    public MonthSummaryDto getMonthSummary(@RequestParam int year, @RequestParam int month) {
        return analyticsService.getMonthSummary(year, month);
    }

    @GetMapping("/category-breakdown")
    public List<CategoryAmountDto> getCategoryBreakdown(@RequestParam int year, @RequestParam int month) {
        return analyticsService.getCategoryBreakdown(YearMonth.of(year, month));
    }

    @GetMapping("/net-worth-trend")
    public List<NetWorthPointDto> getNetWorthTrend(
            @RequestParam("from") LocalDate from, @RequestParam("to") LocalDate to) {
        return analyticsService.getNetWorthTrend(from, to);
    }
}

File: packages/backend/src/modules/analytics/analytics.controller.ts
import { Controller, Get, Query } from '@nestjs/common';
import { ApiTags, ApiQuery } from '@nestjs/swagger';
import { AnalyticsService } from './analytics.service';
import { MonthQueryDto, DateRangeQueryDto } from './dto';

@ApiTags('Analytics')
@Controller('analytics')
export class AnalyticsController {
  constructor(private readonly service: AnalyticsService) {}

  @Get('month-summary')
  @ApiQuery({ name: 'year', required: true, type: Number, description: 'Year (2000-2100)' })
  @ApiQuery({ name: 'month', required: true, type: Number, description: 'Month (1-12)' })
  getMonthSummary(@Query() query: MonthQueryDto) {
    return this.service.getMonthSummary(query.year, query.month);
  }

  @Get('category-breakdown')
  @ApiQuery({ name: 'year', required: true, type: Number, description: 'Year (2000-2100)' })
  @ApiQuery({ name: 'month', required: true, type: Number, description: 'Month (1-12)' })
  getCategoryBreakdown(@Query() query: MonthQueryDto) {
    return this.service.getCategoryBreakdown(query.year, query.month);
  }

  @Get('net-worth-trend')
  @ApiQuery({ name: 'from', required: true, type: String, description: 'Start date (ISO 8601)' })
  @ApiQuery({ name: 'to', required: true, type: String, description: 'End date (ISO 8601)' })
  getNetWorthTrend(@Query() query: DateRangeQueryDto) {
    return this.service.getNetWorthTrend(query.from, query.to);
  }

  @Get('budget-vs-actual')
  @ApiQuery({ name: 'year', required: true, type: Number, description: 'Year (2000-2100)' })
  @ApiQuery({ name: 'month', required: true, type: Number, description: 'Month (1-12)' })
  getBudgetVsActual(@Query() query: MonthQueryDto) {
    return this.service.getBudgetVsActual(query.year, query.month);
  }

  @Get('recurring-costs')
  getRecurringCostSummary() {
    return this.service.getRecurringCostSummary();
  }
}
